packages/stack/proxy/src/proxy.js

```diff
@@ -18,41 +18,46 @@ export class Proxy {
     this.logger = options.logger;
     this.vms = options.vms;
     this.app = null;
-    this.requestManager;
+    this.endpoint = options.endpoint;
+    if (options.endpoint === constants.blockchain.vm) {
+      this.endpoint = this.vms[this.vms.length - 1]();
+    }
+    this.isWs = options.isWs;
   }
 
-  async serve(endpoint, localHost, localPort, ws) {
-    if (endpoint === constants.blockchain.vm) {
-      endpoint = this.vms[this.vms.length - 1]();
-    }
-    this.requestManager = new Web3RequestManager.Manager(endpoint);
+  async nodeReady() {
+    const requestManager = new Web3RequestManager.Manager(this.endpoint);
 
     try {
-      await this.requestManager.send({ method: 'eth_accounts' });
+      await requestManager.send({ method: 'eth_accounts' });
     } catch (e) {
-      throw new Error(__('Unable to connect to the blockchain endpoint'));
+      throw new Error(__(`Unable to connect to the blockchain endpoint on ${this.endpoint}`));
     }
 
+    // disconnect socket connection (NO-OP when HTTP)
+    requestManager.provider.disconnect();
+  }
+
+  async serve(localHost, localPort) {
+
+    await this.nodeReady();
+
     this.app = express();
-    if (ws) {
-      expressWs(this.app);
+    if (this.isWs) {
+      this.wsServer = expressWs(this.app).getWss();
```

Comment:
It doesn't seem like `wsServer` is actually used anywhere. Probably get omit the `getWss()` call?

Reply:
Good catch, thanks. Hangover from testing various options.